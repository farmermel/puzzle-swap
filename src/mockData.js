export const mockPuzzles = [
  {
    title: 'Witch',
    numPieces: '1000',
    piecesMissing: '3-4',
    imgUrl: 'google',
    puzzleId: 4
  },
  {
    title: 'Fairy',
    numPieces: '1000',
    piecesMissing: '3-4',
    imgUrl: 'yahoo',
    puzzleId: 45
  },
  {
    title: 'Unicorn',
    numPieces: '1500',
    piecesMissing: '3-4',
    imgUrl: 'aol',
    puzzleId: 6
  }
]