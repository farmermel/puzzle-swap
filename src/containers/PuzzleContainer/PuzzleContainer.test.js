import React from 'react';
import { shallow } from 'enzyme';
import { PuzzleContainer,
mapStateToProps,
mapDispatchToProps } from './PuzzleContainer';
import { mockPuzzles } from '../../mockData';

describe('PuzzleContainer', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<PuzzleContainer setPuzzles={jest.fn()}
                                       puzzles={mockPuzzles}
                                       history={{push: jest.fn()}} />);
  })

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('componentDidMount', () => {
    it.skip('does some stuff', () => {

    })
  })

  describe('retrievePuzzles', () => {
    it.skip('calls watchData on db with whatever is passed', () => {

    })
  })

  describe('parseChats', () => {
    it('returns an existing chat if ownerId and claimerId match an existing chat', () => {

    })

    it('returns undefined if no chat matches ownderId and claimerId', () => {

    })
  })

  describe('checkForExistingChat', () => {
    it.skip('calls getOnce on db with chats as argument', () => {
        
    })

    it.skip('calls parseChats', () => {

    })

    it.skip('returns an existing chat if there is one', () => {

    })

    it.skip('returns null if there is no existing chat', () => {

    })

    it.skip('catches error if something fails', () => {

    })
  })

  describe('goToChat', () => {
    it('calls push on history object with messages/chatId', () => {
      const existingChat = {
        chatId: 5
      };
      expect(wrapper.instance().props.history.push).not.toHaveBeenCalledWith('messages/5');
      wrapper.instance().goToChat(existingChat);
      expect(wrapper.instance().props.history.push).toHaveBeenCalledWith('messages/5');
    })
  })

  describe('makeNewChat', () => {
    it('calls getFirebaseKey method on db with chats as argument', () => {

    })

    it('makes a postDB object with members, a timestamp, a last message, and a chatId', () => {

    })

    it('calls postUpdate on db with an updates object', () => {

    })

    it('calles checkForExistingChat', () => {

    })

    it('catches error if anything fails', () => {

    })
  })

  describe('handleClaim', () => {
    it('calls checkForExistingChat', () => {

    })

    it('calls goToChat with existingChat if there is already a chat', () => {

    })

    it('calls makeNewChat if there is no existing chat', () => {

    })
  })

  describe('parsePuzzles', () => {
    it.skip('does some stuff', () => {

    })
  })

  describe('displayPuzzles', () => {
    it.skip('does some stuff', () => {

    })
  })

  describe('getImg', () => {
    it.skip('does some stuff', () => {

    })
  })

  describe('mapStateToProps', () => {
    it('maps state to props', () => {
      const mockStore = {
        puzzles: 'are great'
      }
      const mapped = mapStateToProps(mockStore);
      expect(mapped.puzzles).toEqual(mockStore.puzzles);
    })
  })

  describe('mapDispatchToProps', () => {
    it('maps dispatch to props', () => {
      const mockDispatch = jest.fn();
      const mapped = mapDispatchToProps(mockDispatch);
      expect(mockDispatch).not.toHaveBeenCalled();
      mapped.setPuzzles();
      expect(mockDispatch).toHaveBeenCalled();
    })
  })
})