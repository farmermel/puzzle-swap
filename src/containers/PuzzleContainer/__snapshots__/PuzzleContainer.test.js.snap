// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PuzzleContainer matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PuzzleContainer
    history={
        Object {
            "push": [Function],
          }
    }
    puzzles={
        Array [
            Object {
              "imgUrl": "google",
              "numPieces": "1000",
              "piecesMissing": "3-4",
              "puzzleId": 4,
              "title": "Witch",
            },
            Object {
              "imgUrl": "yahoo",
              "numPieces": "1000",
              "piecesMissing": "3-4",
              "puzzleId": 45,
              "title": "Fairy",
            },
            Object {
              "imgUrl": "aol",
              "numPieces": "1500",
              "piecesMissing": "3-4",
              "puzzleId": 6,
              "title": "Unicorn",
            },
          ]
    }
    setPuzzles={[Function]}
    user={
        Object {
            "uid": "5",
            "username": "Casey",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <PuzzleCard
          handleClaim={[Function]}
          puzzle={
                    Object {
                              "imgUrl": "google",
                              "numPieces": "1000",
                              "piecesMissing": "3-4",
                              "puzzleId": 4,
                              "title": "Witch",
                            }
          }
          user={
                    Object {
                              "uid": "5",
                              "username": "Casey",
                            }
          }
/>,
        <PuzzleCard
          handleClaim={[Function]}
          puzzle={
                    Object {
                              "imgUrl": "yahoo",
                              "numPieces": "1000",
                              "piecesMissing": "3-4",
                              "puzzleId": 45,
                              "title": "Fairy",
                            }
          }
          user={
                    Object {
                              "uid": "5",
                              "username": "Casey",
                            }
          }
/>,
        <PuzzleCard
          handleClaim={[Function]}
          puzzle={
                    Object {
                              "imgUrl": "aol",
                              "numPieces": "1500",
                              "piecesMissing": "3-4",
                              "puzzleId": 6,
                              "title": "Unicorn",
                            }
          }
          user={
                    Object {
                              "uid": "5",
                              "username": "Casey",
                            }
          }
/>,
      ],
      "className": "puzzle-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "function",
        "props": Object {
          "handleClaim": [Function],
          "puzzle": Object {
            "imgUrl": "google",
            "numPieces": "1000",
            "piecesMissing": "3-4",
            "puzzleId": 4,
            "title": "Witch",
          },
          "user": Object {
            "uid": "5",
            "username": "Casey",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "45",
        "nodeType": "function",
        "props": Object {
          "handleClaim": [Function],
          "puzzle": Object {
            "imgUrl": "yahoo",
            "numPieces": "1000",
            "piecesMissing": "3-4",
            "puzzleId": 45,
            "title": "Fairy",
          },
          "user": Object {
            "uid": "5",
            "username": "Casey",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "function",
        "props": Object {
          "handleClaim": [Function],
          "puzzle": Object {
            "imgUrl": "aol",
            "numPieces": "1500",
            "piecesMissing": "3-4",
            "puzzleId": 6,
            "title": "Unicorn",
          },
          "user": Object {
            "uid": "5",
            "username": "Casey",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <PuzzleCard
            handleClaim={[Function]}
            puzzle={
                        Object {
                                    "imgUrl": "google",
                                    "numPieces": "1000",
                                    "piecesMissing": "3-4",
                                    "puzzleId": 4,
                                    "title": "Witch",
                                  }
            }
            user={
                        Object {
                                    "uid": "5",
                                    "username": "Casey",
                                  }
            }
/>,
          <PuzzleCard
            handleClaim={[Function]}
            puzzle={
                        Object {
                                    "imgUrl": "yahoo",
                                    "numPieces": "1000",
                                    "piecesMissing": "3-4",
                                    "puzzleId": 45,
                                    "title": "Fairy",
                                  }
            }
            user={
                        Object {
                                    "uid": "5",
                                    "username": "Casey",
                                  }
            }
/>,
          <PuzzleCard
            handleClaim={[Function]}
            puzzle={
                        Object {
                                    "imgUrl": "aol",
                                    "numPieces": "1500",
                                    "piecesMissing": "3-4",
                                    "puzzleId": 6,
                                    "title": "Unicorn",
                                  }
            }
            user={
                        Object {
                                    "uid": "5",
                                    "username": "Casey",
                                  }
            }
/>,
        ],
        "className": "puzzle-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "handleClaim": [Function],
            "puzzle": Object {
              "imgUrl": "google",
              "numPieces": "1000",
              "piecesMissing": "3-4",
              "puzzleId": 4,
              "title": "Witch",
            },
            "user": Object {
              "uid": "5",
              "username": "Casey",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "45",
          "nodeType": "function",
          "props": Object {
            "handleClaim": [Function],
            "puzzle": Object {
              "imgUrl": "yahoo",
              "numPieces": "1000",
              "piecesMissing": "3-4",
              "puzzleId": 45,
              "title": "Fairy",
            },
            "user": Object {
              "uid": "5",
              "username": "Casey",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "function",
          "props": Object {
            "handleClaim": [Function],
            "puzzle": Object {
              "imgUrl": "aol",
              "numPieces": "1500",
              "piecesMissing": "3-4",
              "puzzleId": 6,
              "title": "Unicorn",
            },
            "user": Object {
              "uid": "5",
              "username": "Casey",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
